src/App/useReportAppInitialized.ts

```diff
@@ -0,0 +1,19 @@
+import { useEffect, useState } from 'react';
+
+import { reportExploreMetrics, type ViewName } from 'interactions';
+
+export function useReportAppInitialized() {
+  const [initialized, setInitialized] = useState(false);
+
+  useEffect(() => {
+    if (!initialized) {
+      setInitialized(true);
+
+      const view: ViewName = new URL(window.location.href).searchParams.get('metric')
+        ? 'metric-details'
+        : 'metrics-reducer';
+
+      reportExploreMetrics('app_initialized', { view });
+    }
+  }, [initialized]);
```

Comment:
I think a ref would be nice to use here, instead of state:

```suggestion
import { useEffect, useRef } from 'react';

import { reportExploreMetrics, type ViewName } from 'interactions';

export function useReportAppInitialized() {
  const initialized = useRef(false);

  useEffect(() => {
    if (!initialized.current) {
      initialized.current = true;

      const view: ViewName = new URL(window.location.href).searchParams.get('metric')
        ? 'metric-details'
        : 'metrics-reducer';

      reportExploreMetrics('app_initialized', { view });
    }
  }, []);
```

Using a ref for this helps avoid an unnecessary re-render when `setInitialized(true)` is called.

Reply:
True, thank you Nick! Here it is: https://github.com/grafana/metrics-drilldown/pull/465